File: trackpoint-frontend/src/type/data/projectstat.ts
import type { StatusEnum } from '@/enum'

/**
 * 项目统计中的单个项目
 */
export interface IProjectStat {
  id: string
  name: string
  status: StatusEnum
  create_time: string
  /**
   * 默认事件数量
   */
  default_count: number
  /**
   * 自定义事件数量
   */
  custom_count: number
  /**
   * 记录数量
   */
  record_count: number
  /**
   * 客户端数量
   */
  client_count: number
  /**
   * 所有默认performance事件的总时间，ms
   */
  performance_total_time: number
  /**
   * 所有默认performance事件的js内存占比，单位%
   */
  performance_js_rate: number
  /**
   * 请求报错率，单位%
   */
  request_error_rate: number
  /**
   * js报错次数
   */
  js_error_count: number
}

File: trackpoint-backend/domain/vo/data/project_stat.py


from datetime import datetime
from pydantic import BaseModel
from enums import StatusEnum


class ProjectStatVO(BaseModel):
    """项目统计中响应体中的单个项目"""
    id: str
    name: str
    status: StatusEnum
    create_time:datetime
    default_count: int
    """默认事件数量"""
    custom_count: int
    """自定义事件数量"""
    record_count: int
    """记录数量"""
    client_count: int
    """客户端数量"""
    performance_total_time: float
    """所有默认performance事件的总时间的平均值，单位ms"""
    performance_js_rate: float
    """所有默认performance事件的js内存占比平均值，单位%"""
    request_error_rate: float
    """请求报错率，单位%"""
    js_error_count: int
    """js报错次数"""
